providers/src/airflow/providers/amazon/aws/operators/dms.py

```diff
@@ -277,3 +288,551 @@ def execute(self, context: Context):
+                    self.replication_config_arn,
+                )
+                if self.deferrable:
+                    if current_state.lower() not in self.VALID_STATES:
+                        self.log.info("Deferring until terminal status reached.")
+                        self.defer(
+                            trigger=DmsReplicationTerminalStatusTrigger(
+                                replication_config_arn=self.replication_config_arn,
+                                waiter_delay=self.waiter_delay,
+                                waiter_max_attempts=self.waiter_max_attempts,
+                                aws_conn_id=self.aws_conn_id,
+                            ),
+                            method_name="retry_execution",
+                        )
+                    if provision_status not in self.TERMINAL_PROVISION_STATES:  # not deprovisioned:
+                        self.log.info("Deferring until deprovisioning completes.")
+                        self.defer(
+                            trigger=DmsReplicationDeprovisionedTrigger(
+                                replication_config_arn=self.replication_config_arn,
+                                waiter_delay=self.waiter_delay,
+                                waiter_max_attempts=self.waiter_max_attempts,
+                                aws_conn_id=self.aws_conn_id,
+                            ),
+                            method_name="retry_execution",
+                        )
+
+                else:
+                    self.hook.get_waiter("replication_terminal_status").wait(
+                        Filters=[{"Name": "replication-config-arn", "Values": [self.replication_config_arn]}],
+                        WaiterConfig={"Delay": self.waiter_delay, "MaxAttempts": self.waiter_max_attempts},
+                    )
+                    self.hook.get_waiter("replication_deprovisioned").wait(
+                        Filters=[{"Name": "replication-config-arn", "Values": [self.replication_config_arn]}],
+                        WaiterConfig={"Delay": self.waiter_delay, "MaxAttempts": self.waiter_max_attempts},
+                    )
+                    self.hook.delete_replication_config(self.replication_config_arn)
+                    self.handle_delete_wait()
+
+        else:
+            self.log.info("DMS replication config(%s) does not exist.", self.replication_config_arn)
+
+    def handle_delete_wait(self):
+        if self.deferrable:
+            self.log.info("Deferring until replication config is deleted.")
+            self.defer(
+                trigger=DmsReplicationConfigDeletedTrigger(
+                    replication_config_arn=self.replication_config_arn,
+                    waiter_delay=self.waiter_delay,
+                    waiter_max_attempts=self.waiter_max_attempts,
+                    aws_conn_id=self.aws_conn_id,
+                ),
+                method_name="execute_complete",
+            )
```

Comment:
That should be put under `if self.wait_for_completion:`. If `deferrable` is `True` but `wait_for_completion` is `False`, you should not wait